### DISCUSSION TITLE
Finding duplicate entries?

### DISCUSSION THREAD

**Discussion by Damien455** [msiemens/tinydb#357] [Q&A] [answered]
HI, is there a more efficient way of finding duplicate entries other than outputting db.all() and iterating through the whole lot, please?

**Comment by msiemens** [comment]
Hey @Damien455! One of the main goals of TinyDB is to be as small and simple as possible. Unfortunately, this means that there's no built-in way to check for duplicates.

**Reply by Damien455** [reply]
thank you :)  Love TinyDB for its simplicity and low resources it needs to run

**Comment by therden** [comment, marked as answer]
Hello @Damien455,

You've likely figured this out for yourself at some point over the last 2-3 months, but for fun and practice this afternoon I wrote a small program demonstrating an approach to identifying the `doc_id` of every **TinyDB** record that (other than `doc_id` value) has at least one doppelgänger in the same db file.

The program _does_ iterate through all the records in the database, but I see no way around that.

If interested, see the Github repo at (https://github.com/therden/find_duplicate_TinyDB_records).

**Reply by Damien455** [reply]
Hi, I had been lazy and just make sure duplicates did not occur

so this will be useful

ty

**Comment by therden** [comment]
I'm glad, and you're welcome!
